helm/hpcc/templates/_helpers.tpl

```diff
@@ -429,6 +429,34 @@ Pass in dict with root and secretsCategories
 {{- end -}}
 {{- end -}}
 
+{{/*
+Generate Prometheus scrape annotations
+Enables selfdiscovery of metrics service on configured path/port
+Requires metrics.sinks[type=prometheus]
+Pass in metric config
+*/}}
+{{- define "hpcc.addPrometheusScrapeAnnotations" -}}
+{{- if hasKey . "sinks" }}
+ {{ range $sink := .sinks -}}
+  {{ range $sinkattindex, $sinkatts := $sink -}}
+   {{- if and  (eq $sinkattindex "type") }}
+    {{- if eq $sinkatts "prometheus" }}
+     {{- if hasKey $sink "settings" }}
+      {{- if hasKey $sink.settings "autodiscovery" }}
+       {{- if eq $sink.settings.autodiscovery true }}
+        prometheus.io/scrape: 'true'
+        prometheus.io/path: {{ default "/metrics" $sink.settings.path }}
+        prometheus.io/port: {{ default 8767 $sink.settings.port | quote }}
```

Comment:
Should not indent here (in functions), because that fixes the level of indentation and might not be right for each context it's used in, but instead use ' | indent <n>' where function used. 
Hence in all our other helper functions there is no indent of what they generate.